x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/SparseFileTracker.java

```diff
@@ -49,6 +50,23 @@ public long getLength() {
         return length;
     }
 
+    public List<Tuple<Long, Long>> getCompletedRanges() {
+        List<Tuple<Long, Long>> completedRanges = null;
+        synchronized (mutex) {
+            assert invariant();
+            for (Range range : ranges) {
+                if (range.isPending()) {
+                    continue;
+                }
+                if (completedRanges == null) {
```

Comment:
Random NIT/Question: why use a `LinkedList` for this, seems if we're actually worried about this running so often that it'll be tricky on the GC `ArrayList` would perform more predictably and is what we generally use for this type of spot?

Reply:
Agreed, that's what we usually use, I changed that.